std/experimental/color/rgb.d

```diff
@@ -0,0 +1,852 @@
+    RGBColorSpaceDef!F("SMPTE-C RGB",    &linearToGamma!(F, 2.2), &gammaToLinear!(F, 2.2), WhitePoint!F.D65, xyY!F(0.6300, 0.3400, 0.212395), xyY!F(0.3100, 0.5950, 0.701049), xyY!F(0.1550, 0.0700, 0.086556)),
+    RGBColorSpaceDef!F("Wide Gamut RGB", &linearToGamma!(F, 2.2), &gammaToLinear!(F, 2.2), WhitePoint!F.D50, xyY!F(0.7350, 0.2650, 0.258187), xyY!F(0.1150, 0.8260, 0.724938), xyY!F(0.1570, 0.0180, 0.016875))
++/
+];
+
+template RGBColorSpaceMatrix(RGBColorSpace cs, F)
+{
+    enum F[3] ToXYZ(xyY!F c) = [ c.x/c.y, F(1), (F(1)-c.x-c.y)/c.y ];
+
+    // get the colour space definition
+    enum def = RGBColorSpaceDefs!F[cs];
+    // build a matrix from the 3 colour vectors
+    enum r = def.red, g = def.green, b = def.blue;
+    enum m = transpose([ ToXYZ!r, ToXYZ!g, ToXYZ!b ]);
+
+    // multiply by the whitepoint
+    enum w = [ def.white.tupleof ];
+    enum s = multiply(inverse(m), w);
+
+    // return colourspace matrix (RGB -> XYZ)
+    enum F[3][3] RGBColorSpaceMatrix = [[ m[0][0]*s[0], m[0][1]*s[1], m[0][2]*s[2] ],
+                                        [ m[1][0]*s[0], m[1][1]*s[1], m[1][2]*s[2] ],
+                                        [ m[2][0]*s[0], m[2][1]*s[1], m[2][2]*s[2] ]];
+}
+
+
+package(std.experimental.color):
+
+T linearTosRGB(T)(T s) if(isFloatingPoint!T)
+{
+    if(s <= T(0.0031308))
+        return T(12.92) * s;
+    else
+        return T(1.055) * s^^T(1.0/2.4) - T(0.055);
+}
+T sRGBToLinear(T)(T s) if(isFloatingPoint!T)
+{
+    if(s <= T(0.04045))
+        return s / T(12.92);
+    else
+        return ((s + T(0.055)) / T(1.055))^^T(2.4);
+}
+
+T linearToGamma(T, T gamma)(T v) if(isFloatingPoint!T)
+{
+    return v^^T(1.0/gamma);
+}
+T gammaToLinear(T, T gamma)(T v) if(isFloatingPoint!T)
+{
+    return v^^T(gamma);
+}
+
+T toGrayscale(RGBColorSpace colorSpace = RGBColorSpace.sRGB, T)(T r, T g, T b) pure if(isFloatingPoint!T)
+{
+    // TODO: this function only works for sRGB!
+    // *** others will compile, and produce wrong results!
+/*
+    TODO: do i have this function right? am i in the wrong space?
+
+    // i think this is the linear conversion coefficients
+    wiki = 0.299r + 0.587g + 0.114b
+    ps: (0.3)+(0.59)+(0.11)
+
+    // but gimp source seems to use this one... perhaps this performs an sRGB space estimate, or is THIS the linear one?
+    gimp: 0.21 R + 0.72 G + 0.07 B
+    another = 0.2126 R + 0.7152 G + 0.0722 B   (apparently in linear space?)
+*/
+    return toGamma!colorSpace(T(0.299)*toLinear!colorSpace(r) + T(0.587)*toLinear!colorSpace(g) + T(0.114)*toLinear!colorSpace(b));
+}
+T toGrayscale(RGBColorSpace colorSpace = RGBColorSpace.sRGB, T)(T r, T g, T b) pure if(isIntegral!T)
+{
+    alias F = FloatTypeFor!T;
+    return convertPixelType!T(toGrayscale!colorSpace(convertPixelType!F(r), convertPixelType!F(g), convertPixelType!F(b)));
+}
+
+
+// helpers to parse color components from color component string
+template CanFind(string s, char c)
```

Comment:
Nit, enum templates should be styled like variables (`camelCase`, not `CapitalCase`)